src/win/fs.c

```diff
@@ -1941,6 +1941,110 @@ INLINE static void fs__stat_prepare_path(WCHAR* pathw) {
   }
 }
 
+INLINE static int fs__stat_directory(WCHAR* path, uv_stat_t* statbuf,
+    int do_lstat) {
+
+  HANDLE handle;
+  FILE_STAT_BASIC_INFORMATION stat_info;
+  FILE_FULL_DIR_INFORMATION* pdir_info;
+  IO_STATUS_BLOCK ioStatusBlock;
+  NTSTATUS status;
+  WCHAR buf[32 * 1024];
+  WCHAR* path_dirpath = NULL;
+  WCHAR* path_filename = NULL;
+  BOOL ret_val = FALSE;
+  DWORD ret_error = 0;
+
+  if (uv__fs_split_path(path, &path_dirpath, &path_filename) != 0) {
+    ret_error = GetLastError();
+    goto cleanup;
+  }
+
+  /* Get directory handle */
+  handle = CreateFileW(path_dirpath,
+                       FILE_LIST_DIRECTORY,
+                       FILE_SHARE_READ | FILE_SHARE_WRITE | FILE_SHARE_DELETE,
+                       NULL,
+                       OPEN_EXISTING,
+                       FILE_FLAG_BACKUP_SEMANTICS,
+                       NULL);
+
+  uv__free(path_dirpath);
+  path_dirpath = NULL;
+
+  if (handle == INVALID_HANDLE_VALUE) {
+    ret_error = GetLastError();
+    goto cleanup;
+  }
+
+  do {
+    /* Get files in the directory */
+    status = pNtQueryDirectoryFile(handle,
+                                   NULL,
+                                   0,
+                                   0,
+                                   &ioStatusBlock,
+                                   buf,
+                                   sizeof(buf),
+                                   FileFullDirectoryInformation,
+                                   FALSE,
+                                   NULL,
```

Comment:
The FileName should probably be passed here. Not sure how my comment ended up so far away on a different line

Reply:
I tried your suggestion but got the same error as in the issue (UV__EACCES).